.dev/edge-native-base/src/oauth2_proxy/app.py

```diff
@@ -57,8 +65,9 @@ def oauth_status():
             A 202 status if the user has an auth session
 
         """
-        if NATIVE_APP_MODE == "container" or NATIVE_APP_MODE == "dev":
+        if not HAVE_JUPYTERHUB:
```

Comment:
This changes the authentication to be disabled in a negative condition rather than positively and explicitly. If HAVE_JUPYTERHUB were to be false due to some failure related to the presence of environment variables then the application would be implicitly unprotected.